tests/repository_test.py

```diff
@@ -380,17 +401,56 @@ def test_swift_hook(tempdir_factory, store):
     )
 
 
-def test_golang_hook(tempdir_factory, store):
+def test_golang_system_hook(tempdir_factory, store):
+    with mock.patch.object(golang, '_install_go') as install_go_mck:
+        ret, out = _try_hook_repo(
+            tempdir_factory, store, 'golang_hooks_repo',
+            'golang-hook', [], config_kwargs={
+                'hooks': [{
+                    'id': 'golang-hook',
+                    'language_version': 'system',
+                }],
+            },
+        )
+        assert ret == 0
+        install_go_mck.assert_not_called()
+
+
+def test_golang_default_to_version_hook(tempdir_factory, store):
+    with mock.patch.object(helpers, 'exe_exists') as exe_exists_mck:
+        with mock.patch.object(golang, '_infer_go_version') as _infer_go_mck:
+            exe_exists_mck.return_value = False
+            _infer_go_mck.return_value = '1.18.4'
+            _test_hook_repo(
+                tempdir_factory, store, 'golang_hooks_repo',
+                'golang-hook', [], b'hello world from go1.18.4\n',
+                config_kwargs={
+                    'hooks': [{
+                        'id': 'golang-hook',
+                        'language_version': 'default',
+                    }],
+                },
+            )
+            _infer_go_mck.assert_called_once_with(C.DEFAULT)
```

Comment:
I don't think the mocks are useful here?  can't this just make an assertion based on the real execution and output?  this test is just a reimplementation of the implementation so it's not actually testing anything (same for the mock above as well)

Reply:
done